batch/batch_postgres.go

```diff
@@ -0,0 +1,44 @@
+//go:build kvdb_postgres
+// +build kvdb_postgres
+
+package batch
+
+import (
+	"github.com/lightningnetwork/lnd/kvdb"
+)
+
+// run executes the current batch of requests. It avoids doing actual batching
+// to prevent excessive serialization errors and deadlocks.
+func (b *batch) run() {
+	// Clear the batch from its scheduler, ensuring that no new requests are
+	// added to this batch.
+	b.clear(b)
+
+	// If a cache lock was provided, hold it until the this method returns.
+	// This is critical for ensuring external consistency of the operation,
+	// so that caches don't get out of sync with the on disk state.
+	if b.locker != nil {
+		b.locker.Lock()
+		defer b.locker.Unlock()
+	}
+
+	// Apply each request in the batch in its own transaction. Requests that
+	// fail will be retried by the caller.
+	for _, req := range b.reqs {
```

Comment:
Perhaps we should just bypass the batch scheduler all together for postgres? 

IIRC, we added it originally to speed up initial graph sync for bbolt, by reducing the number of total transactions we did.